### PR TITLE
add some minor tests and comments

### PR DESCRIPTION
**Pull request by ctb**
a first look at some of the functions :)

ready for review, I think.

### CODE REVIEW

File: tests/test_search_utils.py [hunk 1]
```diff
@@ -34,6 +34,42 @@ def test_gen_mh():
...
+
+    lineage_hashD = defaultdict(test_gen_mh)
+
+    add_hashes_at_ranks(lineage_hashD, [hashval1], lin1, 'genus')
+
+    for rank in 'superkingdom', 'phylum', 'class', 'order', 'family', 'genus':
+        check_lin = pop_to_rank(lin1, rank)
+        assert list(lineage_hashD[check_lin].hashes) == [hashval1]
+
+    # CTB: I believe this should NOT work because lin1 is at species level
```
Comment (bluegenes) [inline tests/test_search_utils.py:50]: yep. I was popping to rank outside of this fn, but think it's worth popping to rank here too. I made some changes to the `add_hashes_at_rank` function in master that should fix.

File: tests/test_search_utils.py [hunk 2]
```diff
@@ -34,6 +34,42 @@ def test_gen_mh():
...
+    # but we are only asking to do this down to genus. NTP?
+    assert list(lineage_hashD[lin1].hashes) == [hashval1]
```
Comment (bluegenes) [inline tests/test_search_utils.py:52]:
```suggestion
    assert lin1 not in lineage_hashD.keys()
```

**Comment by bluegenes** [comment]
Thanks!! Ok if I merge these in?

**Comment by ctb** [comment]
> Thanks!! Ok if I merge these in?

of course! but there are now some conflicts :(

**Comment by ctb** [comment]
(note, you can merge THIS PR by checking it out, fixing on local branch, merging there, and then pushing to your master; github will figure it out)

**Pull request merged**